src/v/kafka/client/client.cc

```diff
@@ -174,6 +178,65 @@ ss::future<produce_response::partition> client::produce_record_batch(
       });
 }
 
+ss::future<produce_response> client::produce_records(
+  model::topic topic, std::vector<record_essence> records) {
+    absl::node_hash_map<model::partition_id, storage::record_batch_builder>
+      partition_builders;
+
+    // Assign records to batches per topic_partition
+    for (auto& record : records) {
+        auto p_id = record.partition_id;
+        if (!p_id) {
+            p_id = co_await gated_retry_with_mitigation([&, this]() {
```

Comment:
might passing the lambda (a temporary) to `gated_retry_with_mitigation` trigger the gcc bug on keeping temporaries available in co_await expressions?

Reply:
I've updated to gcc-10.3 and it seems fine. What do I need to do? Pass as lvalue?

Reply:
`gated_retry_with_mitigation` takes the argument by value, which should be ok?

Reply:
yes then it should be ok